Sources/Baggage/BaggageContext.swift

```diff
@@ -41,11 +41,21 @@ public struct BaggageContext {
         get {
             self._storage[ObjectIdentifier(key)]?.forceUnwrap(key)
         } set {
-            self._storage[ObjectIdentifier(key)] = newValue == nil ? nil : ValueContainer(value: newValue!)
+            self._storage[ObjectIdentifier(key)] = newValue == nil ? nil : ValueContainer(keyName: String(describing: key.self), value: newValue!)
         }
     }
 
+    public var printableMetadata: [String: CustomStringConvertible] {
+        var description = [String: CustomStringConvertible]()
+        for (_, value) in self._storage {
+            // TODO: key could be not unique
+            description[value.keyName] = String(describing: value.value)
+        }
+        return description
+    }
```

Comment:
That's too specific to our specific "we need to print those" need IMHO.

It also means that we have:

- to allocate a dictionary which we may not want to do (what if we only are looking for 1 key, no need to look for the rest if we found it)
- we actually render the values, even if we'd never print them -- rendering (`string(describing:`) for them might be expensive (say, storage is integers, but printing them requires base64ing them, would be a waste to make this on every log statement even if we never printed them etc.

What if we get inspired by OpenTracing here: https://github.com/opentracing/opentracing-java/blob/master/opentracing-api/src/main/java/io/opentracing/SpanContext.java#L59

```
    Iterable<Map.Entry<String, String>> baggageItems();
```

so return an iterable with the rendered key and `Any` perhaps in our case?